src/analyses/threadReturn.ml

```diff
@@ -29,9 +29,13 @@ struct
     ctx.local
 
   let enter ctx (lval: lval option) (f:fundec) (args:exp list) : (D.t * D.t) list =
-    [ctx.local, false]
+    if ctx.edge = MyCFG.Skip && ctx.node = MyCFG.dummy_node then
+      (* We are inside enter_with inside a startfun, and thus the current function retruning is the main function *)
+      [ctx.local, true]
```

Comment:
Shouldn't this already be covered by `let startstate v = true`?

Reply:
I see. We even have `morphstate` for some hacky things like this, but that doesn't even help here because it's called before `enter`. This is probably another things that #178 can improve.